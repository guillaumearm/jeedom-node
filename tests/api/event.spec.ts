import { evolve, take } from 'ramda'
import { prepareJeedomTest } from '../utils'

describe('rpc.event', () => {
  const [api, recordTest] = prepareJeedomTest()

  recordTest('event::changes', async () => {
    const changes = await api.event.changes()
    const prepareData = evolve({ result: take(1) })

    expect(prepareData(changes)).toMatchSnapshot()
  })
})
