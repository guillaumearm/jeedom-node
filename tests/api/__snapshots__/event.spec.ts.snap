// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rpc.event event::changes 1`] = `
Object {
  "datetime": 1541448131.741,
  "result": Array [
    Object {
      "datetime": 1541448004.9389,
      "name": "cmd::update",
      "option": Object {
        "cmd_id": "188",
        "collectDate": "2018-11-05 21:00:04",
        "display_value": "15G",
        "value": "15G",
        "valueDate": "2017-10-02 10:02:13",
      },
    },
  ],
}
`;
